src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1779,35 +1853,13 @@ public Task<IOperationResult> EnforceConfigurationAsync(Action<IOperationProgres
                                 }
                             }
 
-                            EnforceRevisionConfigurationCommand cmd;
-                            do
-                            {
-                                token.Delay();
-                                cmd = new EnforceRevisionConfigurationCommand(this, ids, result, includeForceCreated, token);
-                                await _database.TxMerger.Enqueue(cmd);
-                            } while (cmd.MoreWork);
+                            await waitAction(ids, token);
```

Comment:
let's do it a bit differently.
pass a delegate that will create a command, something like this 
```
RevisionsScanningOperationCommand cmd;
do
{
    token.Delay();
    cmd = CreateCommand(ids, token);
    await _database.TxMerger.Enqueue(cmd);
} while (cmd.MoreWork);
```

Reply:
Done